src/main/java/seedu/address/logic/parser/SetConsultationFeeCommandParser.java

```diff
@@ -0,0 +1,47 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import java.math.BigDecimal;
+
+import seedu.address.logic.commands.SetConsultationFeeCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input argument and returns a SetConsultationFeeCommand
+ */
+public class SetConsultationFeeCommandParser implements Parser<SetConsultationFeeCommand> {
+
+    private static final String REGEX = "^\\$?\\d+(\\.\\d{2})?$";
+    /**
+     * Parses the given {@code String} of arguments in the context of the SetConsultationFeeCommand
+     * and returns a SetConsultationFeeCommand object for execution
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public SetConsultationFeeCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        final String[] tokens = trimmedArgs.split("\\s+");
+        // check that there is only 1 argument
+        if (tokens.length != 1) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, SetConsultationFeeCommand.MESSAGE_USAGE));
+        }
+        if (!tokens[0].matches(REGEX)) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, SetConsultationFeeCommand.MESSAGE_USAGE));
+        }
+        String token = tokens[0];
+        if (token.charAt(0) == '$') {
+            token = token.substring(1);
+        }
+        String cents = "00";
```

Comment:
I can not see the need for this part. why not just new BigDecimal(token)

Reply:
The reason I've implemented it this way is because I allowed the doctor to enter in the fee in the form "40." without the cents ($40.00). I wasn't sure if it would be converted properly so I went with this method. 

I've just went to check if simply passing in the token would work and it does work as intended, so I'll be removing this segment. 

Also, I've decided to not allow the doctor to enter the fee in the format stated above anymore as I found it quite pointless.